### ISSUE TITLE
Runtime Checkable Protocol

### ISSUE THREAD

**Issue by karkir0003** [pytorch/pytorch#77196]
### 🐛 Describe the bug

```
Traceback (most recent call last):
  File "C:\Users\karki\OneDrive\Desktop\Deep-Learning-Playground\backend\driver.py", line 63, in <module>
    drive(["nn.Linear(4, 10)", "nn.Linear(10, 1)"], "CELOSS", default=True)
  File "C:\Users\karki\OneDrive\Desktop\Deep-Learning-Playground\backend\driver.py", line 58, in drive
    train_loader, test_loader = get_dataloaders(X_train_tensor, y_train_tensor, X_test_tensor, y_test_tensor, batch_size=20)
  File "C:\Users\karki\OneDrive\Desktop\Deep-Learning-Playground\backend\utils.py", line 17, in get_dataloaders
    train_loader = DataLoader(train, batch_size=batch_size, shuffle=False, drop_last=True)
  File "C:\Users\karki\anaconda3\envs\dlplayground\lib\site-packages\torch\utils\data\dataloader.py", line 200, in __init__
    if isinstance(dataset, IterableDataset):
  File "C:\Users\karki\anaconda3\envs\dlplayground\lib\typing.py", line 1497, in __instancecheck__
    raise TypeError("Instance and class checks can only be used with"
TypeError: Instance and class checks can only be used with @runtime_checkable protocols
```

### Versions

Using python version 3.10. I have the error shown in the traceback when I'm using the following code

```
model = DLModel(parse_user_architecture(user_arch))
    print(f"model: {model}")
    optimizer = get_optimizer(model, "sgd", learning_rate=0.05)
    criterion = LossFunctions[criterion]
    train_loader, test_loader = get_dataloaders(X_train_tensor, y_train_tensor, X_test_tensor, y_test_tensor, batch_size=20)
    train_loss, test_loss = train_model(model, train_loader, test_loader, optimizer, criterion, epochs)
    print(f"train loss: {train_loss}")
    print(f"test loss: {test_loss}")
```

The `get_dataloaders()` function is shown below:
```
from torch.utils.data import TensorDataset, DataLoader

def get_dataloaders(X_train_tensor, y_train_tensor, X_test_tensor, y_test_tensor, batch_size):
    """
    Helper function to get Dataloaders for X_train, y_train, X_test, y_test
    Args:
        X_train_tensor (torch.Tensor)
        y_train_tensor (torch.Tensor)
        X_test_tensor (torch.Tensor)
        y_test_tensor (torch.Tensor)
        batch_size (int)
    """
    train = TensorDataset(X_train_tensor, y_train_tensor)
    test = TensorDataset(X_test_tensor, y_test_tensor)
    print(f"train dims: {train}")
    print(f"test dims: {test}")
    train_loader = DataLoader(train, batch_size=batch_size, shuffle=False, drop_last=True)
    test_loader = DataLoader(test, batch_size=batch_size, shuffle=False, drop_last=True)
    return train_loader, test_loader 
```

Can someone help identify how I can resolve this issue. The parsing of the "user architecture" works perfectly fine. It's just that the DataLoader() stuff is where the error is happening

cc @SsnL @VitalyFedyunin @ejguan @NivekT

**Comment by karkir0003** [comment]
I also am using pytorch version 1.10.2 in a separate conda environment. The yml is shown below:

```
name: dlplayground
channels:
  - conda-forge
  - defaults
dependencies:
  - flake8
  - jupyter
  - matplotlib
  - seaborn
  - mypy
  - numpy
  - pytest
  - python>=3.7
  - pandas
  - scikit-learn
  - boto3
  - pyyaml
  - pip
  - statsmodels
  - pytorch=1.11.0
```

**Comment by karkir0003** [comment]
@ezyang

**Comment by ejguan** [comment]
Hi @karkir0003,
I believe you are using python 3.10 with PyTorch 1.10.
And, this issue has already been fixed in PyTorch 1.11

**Comment by ejguan** [comment]
Closing this issue for now because I believe it's fixed. Feel free to re-open it if you find this is still an issue to you.
Please take reference from https://pytorch.org/get-started/locally/ to see how to install pytorch 1.11
